libs/mve/bundle_io.cc

```diff
@@ -480,4 +483,537 @@ save_photosynther_bundle (Bundle::ConstPtr bundle, std::string const& filename)
+    std::ifstream in_images(images_filename.c_str());
+    check_stream(in_images, images_filename);
+    consume_comment_lines(in_images);
+
+    Bundle::Cameras& bundle_cams = bundle->get_cameras();
+    std::vector<AdditionalCameraInfo> colmap_cams_info;
+    std::string image_line;
+    std::string point_2d_line;
+    double quat[4];
+    math::Matrix3f rot;
+    math::Vec3f trans;
+    int view_id;                    // starts at 0
+    uint32_t view_colmap_id;        // starts at 1
+    uint32_t camera_colmap_id;      // starts at 1
+    int feature_3d_colmap_id;  // starts at 1
+    std::string image_name;
+    while (std::getline(in_images, image_line))
+    {
+        AdditionalCameraInfo colmap_cam_info;
+        CameraInfo bundle_cam;
+        std::stringstream image_line_ss(image_line);
+        image_line_ss >> view_colmap_id
+            >> quat[0] >> quat[1] >> quat[2] >> quat[3]
+            >> trans[0] >> trans[1] >> trans[2]
+            >> camera_colmap_id >> image_name;
+        view_id = view_colmap_id - 1;
+        image_name = util::fs::sanitize_path(image_name);
+        CameraInfo model = camera_colmap_id_to_model.at(camera_colmap_id);
+        bundle_cam.flen = model.flen;
+        bundle_cam.paspect = model.paspect;
+        bundle_cam.ppoint[0] = model.ppoint[0];
+        bundle_cam.ppoint[1] = model.ppoint[1];
+        bundle_cam.dist[0] = model.dist[0];
+        bundle_cam.dist[1] = model.dist[1];
+        rot = get_rot_from_quaternion(quat);
+        std::copy(rot.begin(), rot.end(), bundle_cam.rot);
+        std::copy(trans.begin(), trans.end(), bundle_cam.trans);
+        colmap_cam_info.filename = image_name;
+        colmap_cam_info.radial_distortion = model.dist[0];
+        if (!util::fs::is_absolute(colmap_cam_info.filename))
+            colmap_cam_info.filename = util::fs::join_path(image_path, colmap_cam_info.filename);
+
+        std::string point_2d_line;
+        std::getline(in_images, point_2d_line);
+        std::stringstream point_2d_line_ss(point_2d_line);
+        while (!point_2d_line_ss.eof())
+        {
+            Bundle::Feature2D ref;
+            ref.view_id = view_id;
+            point_2d_line_ss >> ref.pos[0] >> ref.pos[1];
+            point_2d_line_ss >> feature_3d_colmap_id;
+            // A POINT2D that does not belong to a POINT3D has a POINT3D_ID of -1
+            if (feature_3d_colmap_id == -1)
+                ref.feature_id = -1;
+            else
+                ref.feature_id = feature_3d_colmap_id - 1;
+            view_id_to_features_2d[view_id].push_back(ref);
+        }
+        bundle_cams.push_back(bundle_cam);
+        colmap_cams_info.push_back(colmap_cam_info);
+    }    
+    if (camera_info != nullptr)
+        std::swap(*camera_info, colmap_cams_info);
+    in_images.close();
+}
+
+void
+load_colmap_points_3D_txt(std::string const& points3D_filename,
+    std::map< int, std::vector<Bundle::Feature2D> >& view_id_to_features_2d,
```

Comment:
same here

Reply:
`view_id_to_features_2d` is not getting modified within `load_colmap_points_3D_txt()`- only in `load_colmap_images_txt()`. Or am I missing something here?